### PR TITLE
LUCENE-9861: pull tuned vm options into a separate aspect.

### PR DESCRIPTION
**Pull request by dweiss**

### CODE REVIEW

**Review by uschindler — APPROVED** [review]
Looks like a generic way to handle this.

IMHO, the RenderJavadoc tasks should maybe extend JavaExec anyways.

File: gradle/validation/ecj-lint.gradle [hunk 1]
```diff
@@ -74,9 +74,6 @@ allprojects {
         args += [ "-enableJavadoc" ]
         args += [ "-properties", file("${resources}/ecj.javadocs.prefs").absolutePath ]
 
-        // disable for now because individual file excludes cannot be applied to ecj.
```
Comment (rmuir) [inline gradle/validation/ecj-lint.gradle:77]: can we apply the options here as well, like the patch on the JIRA? It was a large win for ecjlint for me.

Comment (rmuir) [reply gradle/validation/ecj-lint.gradle:77]: nm, i see, it must be a "JavaExec" ?

**Comment by rmuir** [comment]
I compared times with another PR (just looking at the github check times), and it shaved a few minutes off

File: gradle/validation/ecj-lint.gradle (hunk 1 above)
Comment (uschindler) [reply gradle/validation/ecj-lint.gradle:77]: It is. That's the trick here. 😜

Comment (dweiss) [reply gradle/validation/ecj-lint.gradle:77]: These vm options are applied to all ecjLint* tasks - I just moved the application to that separate file.

Comment (uschindler) [reply gradle/validation/ecj-lint.gradle:77]: In addition, it's automatically activated for all JavaExec tasks, no need to handle it in all custom tasks.

**Pull request merged**